### PR TITLE
[ASR] Fix GPU memory leak in transcribe_speech.py

### PR DESCRIPTION
**Pull request by rlangman**
# What does this PR do ?

Fix memory leak in utility used by `transcribe_speech.py` when transcribing partial audio segments, by moving logit tensor stored in results to CPU.

**Collection**: [ASR]

# Changelog 
- Modify `transcribe_partial_audio` to move logits to CPU when when `logprobs=False`.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

### CODE REVIEW

**Review by stevehuang52 — APPROVED** [review]
LGTM, thanks~
